### PR TITLE
[ENG-2676] Show operator context on nodes

### PR DESCRIPTION
**Pull request by vsreekanti**
## Describe your changes and why you are making these changes

This PR updates the node design to show context about operators on each node. In particular, it replaces the generic icon at the top of each node with:
1. the engine used by compute operators
2. the data system used by load & save operators.

This PR also takes the opportunity to refactor a bunch of code that was very redundant — we've removed all the duplicate node definitions and collapsed them into a single Node class with a little bit of branching logic.

It also generalizes what was previously called `EngineItem` to `ResourceItem` to also support data systems in addition to compute systems. 

## Related issue number (if any)

ENG-2676, ENG-2648

## Loom demo (if any)

https://www.loom.com/share/2101e9a838e648c592251e36529403e3

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by likawind — COMMENTED** [review]
The code changes LGTM! May leave more comments after looking at the demo

**Comment by vsreekanti** [comment]
@likawind, just added the demo video up above. 🙂  Let me know what you think!

Also, I don't seem to have access to the Aqueduct assets bucket. Could someone please upload the images in `ResourceItem` to that bucket?

File: src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  resourceCustomName?: string; // If included, this will override the default name of the engine that we use.
+  defaultBackgroundColor?: string; // If included, this will override the default background color if none is specified for the particular engine.
+  size?: string; // The font and icon size in pixels.
+  collapseName?: boolean; // If true, the name associated with this engine will be in a tooltip. False by default.
+}
+
+// These two objects are used when we want to show custom definitions for the ResourceItem. Otherwise,
+// we default back to the ServicesLogos imported above.
+const ResourceItemBackgrounds = {
+  [EngineType.Databricks]: '#FF3B29', // The Databricks logo color.
```
Comment (likawind) [inline src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx:19]: any chance we can reuse colors in `theme.platte`?

Comment (vsreekanti) [reply src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx:19]: Not sure we're ever going to be reusing the Databricks logo color anywhere, which is why I hadn't added it to the palette. @agiron123, what do you think?

Comment (agiron123) [reply src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx:19]: @vsreekanti I agree with you here and think that the databricks logo color isn't something that we should add to the theme.

File: src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  [EngineType.K8s]: '#2E6CE6', // The Kubernetes logo color.
+  [EngineType.Lambda]: '#FE7E00', // The Lambda logo color.
+};
+
+const ResourceItemIcons = {
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx:24]: Something we should probably do here is upload these logos to our aqueduct-public-assets-bucket. You never know when these URLs might change out from under us.

File: src/ui/common/src/components/workflows/nodes/Node.tsx [hunk 3]
```diff
@@ -65,22 +99,19 @@ export const Node: React.FC<Props> = ({
   }
 
   let backgroundColor;
-  if (execState?.status === ExecutionStatus.Succeeded) {
+  if (status === ExecutionStatus.Succeeded) {
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/nodes/Node.tsx:102]: We should probably turn this into a switch statement of sorts. maybe even a function of it's own if this is something that we plan to use in other parts of our UI. (Can't think of any at the top of my head right now though 🤔 )

File: src/ui/common/src/components/workflows/nodes/Node.tsx [hunk 4]
```diff
@@ -116,87 +147,173 @@ export const Node: React.FC<Props> = ({
...
-          ? preview
+         * NaN. This only applies to metric operators. */}
+        {!!data.result &&
+        data.spec?.type === OperatorType.Metric &&
+        status === ExecutionStatus.Succeeded
+          ? parseMetricResult(data.result, 3)
           : statusLabels[status]}
       </Typography>
+
+      <Box flex={1} />
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/nodes/Node.tsx:158]: What is the purpose of this box component with nothing inside?

**Review by agiron123 — APPROVED** [review]
Code looks good here. Left a couple comments but you should be able to resolve them and then merge.

**Pull request merged**